tests/conftest.py

```diff
@@ -0,0 +1,60 @@
+import json
+import os
+from contextlib import contextmanager
+from logging import info
+from pathlib import Path
+
+import pytest
+from plumbum import local
+from plumbum.cmd import docker
+
+DOCKER_REPO = os.environ.get("DOCKER_REPO", "docker-socket-proxy")
+IMAGE_NAME = f"{DOCKER_REPO}:local"
+
+
+@pytest.fixture(autouse=True, scope="session")
+def docker_image():
+    """Build local docker image once before starting test suite."""
+    info(f"Building {IMAGE_NAME}...")
+    docker("build", "-t", IMAGE_NAME, Path(__file__).parent.parent)
+    return IMAGE_NAME
```

Comment:
I think that if you're finally doing this because there's no way to reuse the build-and-push action, then you can remove the pre-build step from the CI.

Reply:
Sorry, MB for not removing it. It's gone now, so the image is built in the CI, as you can see from the latest test results. I will just add something in the README explaining that you need to build the image locally before running the tests if you want.